cli/azd/pkg/async/task_test.go

```diff
@@ -0,0 +1,295 @@
+package async
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/test/mocks"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTaskWithResult(t *testing.T) {
+	expectedResult := "result"
+
+	task := NewTask[string]()
+	task.Run(func(ctx *TaskContext[string]) {
+		time.Sleep(250 * time.Millisecond)
+		ctx.SetResult(expectedResult)
```

Comment:
I see..  You are delegating the responsibility to set the Task Result to the `func` dispatched by the task...  

Have you considered updating the `func` signature to return `R` , so the task can use the return value to set the Result instead?

Reply:
A simplified return value would be ideal but it would also limit usage of go functions that we already require for provisioning.